src/build_helper.rs

```diff
@@ -0,0 +1,262 @@
+use core::ops::Range;
+
+use alloc::vec::Vec;
+
+use crate::errors::{DaachorseError, Result};
+
+const INVALID_IDX: u32 = u32::MAX;
+
+/// Helper class in double-array construction to maintain indices of vacant elements and
+/// unused BASE values.
+///
+/// This class manages array elements in fixed-size blocks and supports to extend the array
+/// block by block. Given a constant parameter N, this class maintains information for only
+/// the last N blocks and drops the others according to array extension. We call such last
+/// blocks *active blocks*.
+#[derive(Default)]
```

Comment:
This struct should not imprement Default because `capacity` and `block_len` must not 0.

Reply:
https://github.com/daac-tools/daachorse/pull/48/commits/9640a53fd49bc8a1868b510e970f875ea1e4ffbe